src/engine/filters/enginefilterdelay.h

```diff
@@ -29,6 +29,13 @@ class EngineFilterDelay : public EngineObjectConstIn {
 
     void setDelay(unsigned int delaySamples) {
         m_delaySamples = delaySamples;
+
+        // When mixxx will support other channel count variants than stereo,
+        // the kMaxDelay has to be divisible by the number of channels.
+        // Otherwise, channels may be swapped.
+        VERIFY_OR_DEBUG_ASSERT(m_delaySamples <= SIZE - mixxx::kEngineChannelCount) {
```

Comment:
Is this condition correct, or do we need to substract the engine  buffer size?

Reply:
Can you please explain more about why you think of substracting the engine buffer size instead and which "engine buffer size" you think of? The size of the input buffer? The reason, why the number of channels is subtracted is, that the delay size can't be equal to the size of the delay buffer (m_buf). It can be smaller and subtracting the number of channels ensures, that the channels won't be swapped between. IMO the assertion shouldn't go wrong as original, because it doesn't allow the delay values greater than possible (greater or equal to buffer size), so the `m_delaySamples` values from the examples wouldn't be allowed.

Reply:
Ah sorry, I have misread the code. In case of 0 delay, you but one frame into the buffer and read it back immediately. 
That is why `- mixxx::kEngineChannelCount` is sufficient.

Reply:
No problem. Perfect.